pkg/router/controller/controller.go

```diff
@@ -57,6 +60,24 @@ func (c *RouterController) Run() {
 	if c.WatchNodes {
 		go utilwait.Forever(c.HandleNode, 0)
 	}
+	go c.watchForFirstSync()
+}
+
+// watchForFirstSync signals the router when it sees that the various
+// watchers have successfully listed data from the api.
+func (c *RouterController) watchForFirstSync() {
+	for {
+		syncedAtLeastOnce := c.RoutesListSuccessfulAtLeastOnce() && c.EndpointsListSuccessfulAtLeastOnce() &&
+			(c.Namespaces == nil || c.filteredByNamespace)
+		if syncedAtLeastOnce {
+			if err := c.Plugin.SetSyncedAtLeastOnce(); err != nil {
+				utilruntime.HandleError(err)
+			} else {
+				return
+			}
+		}
+		time.Sleep(1)
```

Comment:
Good call, I keep forgetting that ``time.Sleep()`` takes a duration not an int.  My intention was to sleep for 1s.